[DXCDT-194] Generate docs with tfplugindocs (#269)
auth0/terraform-provider-auth0: docs/data-sources/client.md

---
page_title: "Data Source: auth0_client"
description: |-
  Data source to retrieve a specific Auth0 Application client by 'client_id' or 'name'.
---

# Data Source: auth0_client

Data source to retrieve a specific Auth0 Application client by 'client_id' or 'name'.

## Example Usage

```terraform
# An Auth0 Client loaded using its name.
data "auth0_client" "some-client-by-name" {
  name = "Name of my Application"
}

# An Auth0 Client loaded using its ID.
data "auth0_client" "some-client-by-id" {
  client_id = "abcdefghkijklmnopqrstuvwxyz0123456789"
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Optional

- `client_id` (String) The ID of the client. If not provided, `name` must be set.
- `name` (String) The name of the client. If not provided, `client_id` must be set.

### Read-Only

> - `addons` (List of Object) Addons enabled for this client and their associated configurations. (see [below for nested schema](#nestedatt--addons))

Reviewer: ```suggestion
- `addons` (List of Object) Addons enabled for this client and their associated configurations (see [below for nested schema](#nestedatt--addons)).
```

- `allowed_clients` (List of String) List of applications ID's that will be allowed to make delegation request. By default, all applications will be allowed.
- `allowed_logout_urls` (List of String) URLs that Auth0 may redirect to after logout.
- `allowed_origins` (List of String) URLs that represent valid origins for cross-origin resource sharing. By default, all your callback URLs will be allowed.
- `app_type` (String) Type of application the client represents.
- `callbacks` (List of String) URLs that Auth0 may call back to after a user authenticates for the client. Make sure to specify the protocol (https://) otherwise the callback may fail in some cases. With the exception of custom URI schemes for native clients, all callbacks should use protocol https://.

> - `client_metadata` (Map of String) Metadata associated with the client, in the form of an object with string values (max 255 chars). Maximum of 10 metadata properties allowed. Field names (max 255 chars) are alphanumeric and may only include the following special characters: :,-+=_*?"/\()<>@ [Tab] [Space]

Reviewer: ```suggestion
- `client_metadata` (Map of String) Metadata associated with the client, in the form of an object with string values (max 255 chars). Maximum of 10 metadata properties allowed. Field names (max 255 chars) are alphanumeric and may only include the following special characters: `:,-+=_*?"/\()<>@ [Tab] [Space]`.
```

- `client_secret` (String) Secret for the client. Keep this private. To access this attribute you need to add the `read:client_keys` scope to the Terraform client. Otherwise, the attribute will contain an empty string.
- `cross_origin_auth` (Boolean) Whether this client can be used to make cross-origin authentication requests (true) or it is not allowed to make such requests (false).

> - `cross_origin_loc` (String) URL of the location in your site where the cross-origin verification takes place for the cross-origin auth flow when performing Auth in your own domain instead of Auth0 hosted login page.

Reviewer: ```suggestion
- `cross_origin_loc` (String) URL of the location in your site where the cross-origin verification takes place for the cross-origin auth flow when performing authentication in your own domain instead of Auth0 Universal Login page.
```

- `custom_login_page` (String) The content (HTML, CSS, JS) of the custom login page.
- `custom_login_page_on` (Boolean) Indicates whether a custom login page is to be used.
- `description` (String) Description of the purpose of the client.

> - `encryption_key` (Map of String) Encryption used for Ws-Fed responses with this client.

Reviewer: ```suggestion
- `encryption_key` (Map of String) Encryption used for WS-Fed responses with this client.
```

- `form_template` (String) HTML form template to be used for WS-Federation.
- `grant_types` (List of String) Types of grants that this client is authorized to use.
- `id` (String) The ID of this resource.

> - `initiate_login_uri` (String) Initiate login URI, must be https.

Reviewer: ```suggestion
- `initiate_login_uri` (String) Initiate login URI, must be HTTPS.
```

- `is_first_party` (Boolean) Indicates whether this client is a first-party client.
- `is_token_endpoint_ip_header_trusted` (Boolean) Indicates whether the token endpoint IP header is trusted.

> - `jwt_configuration` (List of Object) Configuration settings for the JWTs issued for this client. (see [below for nested schema](#nestedatt--jwt_configuration))

Reviewer: ```suggestion
- `jwt_configuration` (List of Object) Configuration settings for the JWTs issued for this client (see [below for nested schema](#nestedatt--jwt_configuration)).
```

- `logo_uri` (String) URL of the logo for the client. Recommended size is 150px x 150px. If none is set, the default badge for the application type will be shown.

> - `mobile` (List of Object) Additional configuration for native mobile apps. (see [below for nested schema](#nestedatt--mobile))

Reviewer: ```suggestion
- `mobile` (List of Object) Additional configuration for native mobile apps (see [below for nested schema](#nestedatt--mobile)).
```

> - `native_social_login` (List of Object) Configuration settings to toggle native social login for mobile native applications. Once this is set it must stay set, with both resources set to `false` in order to change the `app_type`. (see [below for nested schema](#nestedatt--native_social_login))

Reviewer: ```suggestion
- `native_social_login` (List of Object) Configuration settings to toggle native social login for mobile native applications. Once this is set it must stay set, with both resources set to `false` in order to change the `app_type` (see [below for nested schema](#nestedatt--native_social_login)).
```

- `oidc_conformant` (Boolean) Indicates whether this client will conform to strict OIDC specifications.
- `organization_require_behavior` (String) Defines how to proceed during an authentication transaction when `organization_usage = "require"`. Can be `no_prompt` (default) or `pre_login_prompt`.
- `organization_usage` (String) Defines how to proceed during an authentication transaction with regards to an organization. Can be `deny` (default), `allow` or `require`.

> - `refresh_token` (List of Object) Configuration settings for the refresh tokens issued for this client. (see [below for nested schema](#nestedatt--refresh_token))

Reviewer: ```suggestion
- `refresh_token` (List of Object) Configuration settings for the refresh tokens issued for this client (see [below for nested schema](#nestedatt--refresh_token)).
```

- `signing_keys` (List of Map of String) List containing a map of the public cert of the signing key and the public cert of the signing key in PKCS7.

> - `sso` (Boolean) Applies only to SSO clients and determines whether Auth0 will handle Single Sign On (true) or whether the identity provider will (false).

Reviewer: ```suggestion
- `sso` (Boolean) Applies only to SSO clients and determines whether Auth0 will handle Single Sign-On (true) or whether the identity provider will (false).
```
